js/client.js

```diff
@@ -835,12 +835,25 @@ function toId() {
 			// querySelector dates back to IE8 so we can use it
 			// fortunate, because form serialization is a HUGE MESS in older browsers
 			var elements = form.querySelectorAll('input[name], select[name], textarea[name], keygen[name]');
-			var out = [];
+			var out = [], checkboxes = {}, radios = {};
 			for (var i = 0; i < elements.length; i++) {
 				var element = elements[i];
-				// TODO: values are a mess in the DOM; checkboxes/select probably need special handling
-				out.push([element.name, element.value]);
-			}
+				if (element.type === 'radio') {
+					var name = ' ' + element.name;
+					if (element.checked) radios[name] = element.value;
+					else if (!radios[name]) radios[name] = false;
+				} else if (element.type === 'checkbox') {
+					if (!checkboxes[' ' + element.name]) checkboxes[' ' + element.name] = [];
+					if (element.checked) checkboxes[' ' + element.name].push(element.value);
```

Comment:
Oh, I see, you're trying to handle the case where multiple checkboxes have the same name.

Is that actually how the spec handles it?

Reply:
Okay I have no idea what I was thinking when I wrote the checkboxes, sorry
Lemme rewrite this, gimme a minute